src/m365/spo/commands/file/file-copy.ts

```diff
@@ -170,59 +174,44 @@ class SpoFileCopyCommand extends SpoCommand {
-      const targetFolderAbsoluteUrl: string = urlUtil.urlCombine(tenantUrl, targetUrl);
-
-      // since the target WebFullUrl is unknown we can use getRequestDigestForSite
-      // to get it from target folder absolute url.
-      // Similar approach used here Microsoft.SharePoint.Client.Web.WebUrlFromFolderUrlDirect
-      spo
-        .getRequestDigest(targetFolderAbsoluteUrl)
-        .then((contextResponse: ContextInfo): void => {
-          if (this.debug) {
-            logger.logToStderr(`contextResponse.WebFullUrl: ${contextResponse.WebFullUrl}`);
-          }
-
-          if (targetUrl.charAt(0) !== '/') {
-            targetUrl = `/${targetUrl}`;
-          }
-          if (targetUrl.lastIndexOf('/') !== targetUrl.length - 1) {
-            targetUrl = `${targetUrl}/`;
-          }
-
-          const requestUrl: string = `${contextResponse.WebFullUrl}/_api/web/GetFileByServerRelativeUrl('${encodeURIComponent(`${targetUrl}${filename}`)}')/recycle()`;
-          const requestOptions: any = {
-            url: requestUrl,
-            method: 'POST',
-            headers: {
-              'X-HTTP-Method': 'DELETE',
-              'If-Match': '*',
-              'accept': 'application/json;odata=nometadata'
-            },
-            responseType: 'json'
-          };
-
-          request.post(requestOptions)
-            .then((): void => {
-              resolve();
-            })
-            .catch((err: any): any => {
-              if (err.statusCode === 404) {
-                // file does not exist so can proceed
-                return resolve();
-              }
-
-              if (this.debug) {
-                logger.logToStderr(`recycleFile error...`);
-                logger.logToStderr(err);
-              }
-
-              reject(err);
-            });
-        }, (e: any) => reject(e));
-    });
+  private recycleFile(tenantUrl: string, targetUrl: string, filename: string, logger: Logger): Promise<void | CommandOutput> {
+    const targetFolderAbsoluteUrl: string = urlUtil.urlCombine(tenantUrl, targetUrl);
+
+    // since the target WebFullUrl is unknown we can use getRequestDigestForSite
+    // to get it from target folder absolute url.
+    // Similar approach used here Microsoft.SharePoint.Client.Web.WebUrlFromFolderUrlDirect
+    return this.getWebFullUrl(targetFolderAbsoluteUrl, logger)
+      .then((webFullUrl: string) => {
+        const targetFileServerRelativeUrl: string = `${urlUtil.getServerRelativePath(webFullUrl, targetUrl)}/${filename}`;
+        const removeOptions: SpoFileRemoveOptions = {
+          webUrl: webFullUrl,
+          url: targetFileServerRelativeUrl,
+          recycle: true,
+          confirm: true,
+          debug: this.debug,
+          verbose: this.verbose
+        };
+        
+        return Cli.executeCommandWithOutput(removeCommand as Command, { options: { ...removeOptions, _: [] } })
+          .catch((err: FileDeleteError) => {
+            logger.logToStderr(err);
```

Comment:
Since we're ignoring the error for some messages, we shouldn't log the output to error here, but rather let it be handled by the caller who'll get full command output.